UefiTestingPkg/FunctionalSystemTests/MemoryProtectionTest/Readme.md

```diff
@@ -7,15 +7,47 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
 
 ## About This Test
 
-This is set of tests to ensure that heap guard, stack guard, null pointer detection, and nx protections are working properly.
+This test verifies the memory protection features of Project Mu. The test checks the
+memory protection policy of the system and adds a test for each memory type of each
+protection type active. For example, if the system protection policy has NX protection
+on EfiBootServicesData, the test will allocate a buffer of that type and check the
+attributes of that buffer either by attempting to execute code within it or by fetching
+the attributes using the memory attribute protocol. The test has a few methods of running.
 
-It consists of:
+## Testing using Resets
 
-- An SMM driver
-- A Shell-based test app
+This method is the slowest, reaching upwards of 45 minutes with a strict protection policy
+(benchmarked on a Q35 DEBUG build). The test will intentionally cause faults for each
+protection policy and use a custom installed fault handler for IA32_PAGE_FAULT exceptions

```

Comment:
I'm assuming `IA32_PAGE_FAULT` is not what the fault handler is on ARM64?

Reply:
@TaylorBeebe - your comment confuses me.  You say that there is no handler for "ARM platforms" but your comment here says that it can be run on ARM64 platforms.  Does the default synchronous handler just reset the system and thus it works?

Reply:
It can be run on ARM64 platforms using the memory attribute protocol. Currently, the test doesn't install a synchronous handler for running the reset method on ARM platforms. It's possible, though.

I don't know of any flag on x86 architectures that indicates we should do a reboot instead of deadloop. We've accomplished the reboot by installing our own interrupt handler within this test, and also the memory protection exception handler used on Mu based platforms does a reboot instead of deadloop.

Reply:
@TaylorBeebe if I understand correctly, since this is the reset section, perhaps you should replace _"This method can be run on either X64 or ARM64 platforms."_ with _"This method can be run as implemented on X64. Currently, the test doesn't install a synchronous handler for initiating a reset on ARM platforms."_

Reply:
My mistake, the final sentence of this section shouldn't list support for ARM platforms. I'll replace it with Michael's suggestion.